packages/frontend/src/components/NavBar/index.tsx

```diff
@@ -1,213 +1,111 @@
-                            `/projects/${projectUuid}/dashboards/${dashboardUuid}/${
-                                savedQueryUuid ? 'view' : 'edit'
-                            }`,
-                        );
-                    }}
-                    size="xs"
-                    ml="md"
-                    variant="white"
-                    compact
-                >
-                    Cancel
-                </Button>
-            </Tooltip>
-        </Center>
+    const isEditingDashboardChart = getIsEditingDashboardChart();
+
+    const navBarMode = useMemo(
+        () =>
+            isEditingDashboardChart
+                ? NavBarMode.EDITING_DASHBOARD_CHART
+                : NavBarMode.DEFAULT,
+        [isEditingDashboardChart],
     );
+
+    return { navBarMode };
 };
 
 const NavBar = memo(() => {
     const { projectUuid } = useParams<{ projectUuid: string }>();
     const { data: projects } = useProjects();
     const { activeProjectUuid, isLoading: isLoadingActiveProject } =
         useActiveProjectUuid({ refetchOnMount: true });
-    const {
-        getIsEditingDashboardChart,
-        getEditingDashboardInfo,
-        clearDashboardStorage,
-    } = useDashboardStorage();
-
     const { isFullscreen } = useApp();
 
-    const dashboardInfo = getEditingDashboardInfo();
-
-    const homeUrl = activeProjectUuid
-        ? `/projects/${activeProjectUuid}/home`
-        : '/';
+    const { navBarMode } = useNavBarMode();
 
     const isCurrentProjectPreview = !!projects?.find(
         (project) =>
             project.projectUuid === activeProjectUuid &&
             project.type === ProjectType.PREVIEW,
     );
 
-    useEffect(() => {
-        window.addEventListener('unload', clearDashboardStorage);
-        return () =>
-            window.removeEventListener('unload', clearDashboardStorage);
-    }, [clearDashboardStorage]);
-
-    if (getIsEditingDashboardChart()) {
-        return (
-            <DashboardExplorerBanner
-                dashboardName={dashboardInfo.name || ''}
-                projectUuid={projectUuid}
-                dashboardUuid={dashboardInfo.dashboardUuid || ''}
-            />
-        );
-    }
-
+    const getHeaderStyles = useCallback(
+        (theme: MantineTheme) => ({
+            ...defaultNavbarStyles,
+            ...(navBarMode === NavBarMode.EDITING_DASHBOARD_CHART && {
+                justifyContent: 'center',
+                borderBottom: 'none',
+                backgroundColor: theme.colors.blue['6'],
+                color: 'white',
+            }),
+        }),
+        [navBarMode],
+    );
     const headerContainerHeight =
         NAVBAR_HEIGHT + (isCurrentProjectPreview ? BANNER_HEIGHT : 0);
 
+    const renderNavBarContent = () => {
+        switch (navBarMode) {
+            case NavBarMode.EDITING_DASHBOARD_CHART:
+                return <DashboardExplorerBanner projectUuid={projectUuid} />;
+            case NavBarMode.DEFAULT:
+            default:
```

Comment:
navbarMode can never be undefined so you can move the content from `default` block to NavbarMode.DEFAULT and call `assertUnreachable` inside `default` block

Reply:
Yes 🤦 
Thanks! Will add this 👍

Reply:
perfect 👍